ports/rp2/rp2_pio.c

```diff
@@ -872,6 +893,7 @@ static mp_obj_t rp2_state_machine_irq(size_t n_args, const mp_obj_t *pos_args, m
         }
 
         if (self->pio->inte0) {
+            rp2_pio_irq_set_exclusive_handler(self->pio, self->pio == pio0 ? PIO0_IRQ_0 : PIO1_IRQ_0);
```

Comment:
Couldn't the second argument here just be `self->irq`?

Reply:
Good spot! Thank you. No idea how I managed to do this two different ways.

Your thorough reviews are much appreciated!